bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/connection/PhilipsAirCipher.java

```diff
@@ -0,0 +1,150 @@
+import java.util.Base64;
+import java.util.Random;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.util.HexUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Performs message de- and encyrption
+ *
+ * @author Michał Boroński - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class PhilipsAirCipher {
+    private static final String AA = "AA";
+
+    private final Logger logger = LoggerFactory.getLogger(PhilipsAirCipher.class);
+
+    private static final BigInteger G = new BigInteger(
+            "A4D1CBD5C3FD34126765A442EFB99905F8104DD258AC507FD6406CFF14266D31266FEA1E5C41564B777E690F5504F213160217B4B01B886A5E91547F9E2749F4D7FBD7D3B9A92EE1909D0D2263F80A76A6A24C087A091F531DBF0A0169B6A28AD662A4D18E73AFA32D779D5918D08BC8858F4DCEF97C2A24855E6EEB22B3B2E5",
+            16);
+    private static final BigInteger P = new BigInteger(
+            "B10B8F96A080E01DDE92DE5EAE5D54EC52C99FBCFB06A3C69A6A9DCA52D23B616073E28675A23D189838EF1E2EE652C013ECB4AEA906112324975C3CD49B83BFACCBDD7D90C4BD7098488E9C219A73724EFFD6FAE5644738FAA31A4FF55BCCC0A151AF5F0DC8B4BD45BF37DF365C1A65E68CFDA76D4DA708DF1FB2BC2E4A4371",
+            16);
+    private static final Random RAND = new Random();
+
+    @Nullable
+    private Cipher decipher;
+    @Nullable
+    private Cipher cipher;
+    private final BigInteger a;
+    private final BigInteger aPow;
+    private static final Charset ASCII_CHARSET = Charset.forName("US-ASCII");
+
+    public PhilipsAirCipher() throws GeneralSecurityException {
+        // prepare numbers for key
+        a = randomForBitsNonZero(256, RAND);
+        aPow = G.modPow(a, P);
+    }
+
+    @SuppressWarnings("null")
+    public void initKey(String key) throws GeneralSecurityException {
+        try {
+            decipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+            decipher.init(Cipher.DECRYPT_MODE, new SecretKeySpec(HexUtils.hexToBytes(key), "AES"),
+                    new IvParameterSpec(new byte[16]));
+
+            cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+            cipher.init(Cipher.ENCRYPT_MODE, new SecretKeySpec(HexUtils.hexToBytes(key), "AES"),
+                    new IvParameterSpec(new byte[16]));
+        } catch (GeneralSecurityException e) {
+            logger.warn("An exception occured", e);
+            cipher = null;
+            decipher = null;
+            throw e;
+        }
+    }
+
+    public String getApow() {
+        return aPow.toString(16);
+    }
+
+    private static BigInteger randomForBitsNonZero(int numBits, Random r) {

```

Comment:
No need to pass in Random as a parameter when you can access the field directly.

Reply:
Correct